php-checks/src/main/java/org/sonar/php/checks/UnusedFunctionParametersCheck.java

```diff
@@ -41,30 +45,56 @@
 
   public static final String KEY = "S1172";
   private static final String MESSAGE = "Remove the unused function parameter \"%s\".";
+  Deque<Boolean> hasFuncGetArgsStack = new ArrayDeque<>();
+
+  /*
+   * TODO: Using the stack can be avoided when the new symbol table will be improved. In particular, when FunctionExpressionTreeImpl will
+   * implement the HasMethodSymbol interface.
+   */
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String callee = SourceBuilder.build(tree.callee()).trim();
+    if (callee.equals("func_get_args")) {
+      hasFuncGetArgsStack.pop();
+      hasFuncGetArgsStack.push(true);
+    }
+    super.visitFunctionCall(tree);
+  }
 
   @Override
   public void visitFunctionDeclaration(FunctionDeclarationTree tree) {
-    checkParameters(tree);
+    hasFuncGetArgsStack.push(false);
     super.visitFunctionDeclaration(tree);
+    boolean hasFuncGetArgs = hasFuncGetArgsStack.pop();
+    if (!hasFuncGetArgs) {
```

Comment:
Is it necessary to assign a variable for this? I would say that the name of the stack indicates what is being checked.